backend/layers/persistence/persistence.py

```diff
@@ -507,6 +507,17 @@ def add_collection_version(self, collection_id: CollectionId) -> CollectionVersi
             session.add(new_version)
             return CollectionVersionId(new_version_id)
 
+    def delete_collection(self, collection_id: CollectionId) -> None:
+        """
+        Delete an unpublished Collection
```

Comment:
what happens to all of the rows and table associated with the collection?

Reply:
Since dataset_version depends on a dataset, which depend on a collection_version, which depend on a collection, deleting a collection should be an atomic database operation. If there isn't a ticket filed for this already, please create and add one for better database session management.

Reply:
Another option it to add a session parameter to these database operations and use the existing session if it's passed in. This might be a heavy lift, I'll try an prototype it.

Reply:
We should leave these sorts of reorg questions for the Redesign 1.1 work. This is what we've been discussing wrt fixing the persistence/business layer interactions.

You can make a ticket if you like and add it to Redesign 1.1.

Reply:
Add a todo to this function to make that change in the future. It's important we don't forget to make this change.